exporter/splunkhecexporter/buffer.go

```diff
@@ -113,8 +125,19 @@ func (c *cancellableGzipWriter) Reset() {
 	c.rawLen = 0
 }
 
+func (c *cancellableGzipWriter) Flush() error {
+	return c.innerWriter.Flush()
+}
+
+func (c *cancellableGzipWriter) OnClose(onClose func() error) {
+	c.onClose = onClose
+}
+
 func (c *cancellableGzipWriter) Close() error {
-	return c.innerWriter.Close()
+	if err := c.innerWriter.Close(); err != nil {
+		return err
+	}
+	return c.onClose()
```

Comment:
`onClose` is nullable, so that call could result in a panic in case no closing hook has been attached